include/libdnf/conf/config_main.hpp

```diff
@@ -0,0 +1,187 @@
+    OptionString & installroot();
+    OptionString & config_file_path();
+    OptionBool & plugins();
+    OptionStringList & pluginpath();
+    OptionStringList & pluginconfpath();
+    OptionString & persistdir();
+    OptionBool & transformdb();
+    OptionNumber<std::int32_t> & recent();
+    OptionBool & reset_nice();
+    OptionString & system_cachedir();
+    OptionBool & cacheonly();
+    OptionBool & keepcache();
+    OptionString & logdir();
+    OptionNumber<std::int32_t> & log_size();
+    OptionNumber<std::int32_t> & log_rotate();
+    OptionStringList & varsdir();
+    OptionStringList & reposdir();
+    OptionBool & debug_solver();
+    OptionStringList & installonlypkgs();
+    OptionStringList & group_package_types();
+
+    /*  NOTE: If you set this to 2, then because it keeps the current
+    kernel it means if you ever install an "old" kernel it'll get rid
+    of the newest one so you probably want to use 3 as a minimum
+    ... if you turn it on. */
+    OptionNumber<std::uint32_t> & installonly_limit();
+
+    OptionStringList & tsflags();
+    OptionBool & assumeyes();
+    OptionBool & assumeno();
+    OptionBool & check_config_file_age();
+    OptionBool & defaultyes();
+    OptionBool & diskspacecheck();
+    OptionBool & localpkg_gpgcheck();
+    OptionBool & gpgkey_dns_verification();
+    OptionBool & obsoletes();
+    OptionBool & showdupesfromrepos();
+    OptionBool & exit_on_lock();
+    OptionSeconds & metadata_timer_sync();
+    OptionStringList & disable_excludes();
+    OptionEnum<std::string> & multilib_policy();  // :api
+    OptionBool & best();                          // :api
+    OptionBool & install_weak_deps();
+    OptionString & bugtracker_url();
+    OptionBool & zchunk();
+    OptionEnum<std::string> & color();
+    OptionString & color_list_installed_older();
+    OptionString & color_list_installed_newer();
+    OptionString & color_list_installed_reinstall();
+    OptionString & color_list_installed_extra();
+    OptionString & color_list_available_upgrade();
+    OptionString & color_list_available_downgrade();
+    OptionString & color_list_available_reinstall();
+    OptionString & color_list_available_install();
+    OptionString & color_update_installed();
+    OptionString & color_update_local();
+    OptionString & color_update_remote();
+    OptionString & color_search_match();
+    OptionBool & history_record();
+    OptionStringList & history_record_packages();
+    OptionString & rpmverbosity();
+    OptionBool & strict();                    // :api
+    OptionBool & skip_broken();               // :yum-compatibility
+    OptionBool & autocheck_running_kernel();  // :yum-compatibility
+    OptionBool & clean_requirements_on_remove();
+    OptionEnum<std::string> & history_list_view();
+    OptionBool & upgrade_group_objects_upgrade();
+    OptionPath & destdir();
+    OptionString & comment();
+    OptionBool & downloadonly();
+    OptionBool & ignorearch();
```

Comment:
Let's go ahead and add the sticky vendor options that are part of #907.

Reply:
We have to first finish review of #907.
We would like to develop dnf5 into state of libdnf master, then we will backport all additional patches from dnf-4